python/taichi/lang/ast/ast_transformer.py

```diff
@@ -18,6 +18,53 @@ def build_Name(ctx, node):
         node.ptr = ctx.get_var_by_name(node.id)
         return node
 
+    @staticmethod
+    def build_AnnAssign(ctx, node):
+        annotation = getattr(node.annotation, 'attr')
+
+        node.value = build_stmt(ctx, node.value)
+        node.target = build_stmt(ctx, node.target)
+
+        is_static_assign = isinstance(
+            node.value, ast.Call) and ASTResolver.resolve_to(
+                node.value.func, ti.static, globals())
+
+        node.ptr = ASTTransformer.build_ann_assign_basic(
+            ctx, node.target, node.value.ptr, is_static_assign, annotation)
+        return node
+
+    @staticmethod
+    def build_ann_assign_basic(ctx, target, value, is_static_assign,
+                               annotation):
+        """Build basic assginment like this: target : annotation = value.
+
+         Args:
+            ctx (ast_builder_utils.BuilderContext): The builder context.
+            target (ast.Name): A variable name. `target.id` holds the name as
+            a string.
+            annotation: A type we hope to assign to the target
+            value: A node representing the value.
+            is_static_assign: A boolean value indicating whether this is a static assignment
+        """
+        is_local = isinstance(target, ast.Name)
+        if is_static_assign:
+            if not is_local:
+                raise TaichiSyntaxError(
+                    "Static assign cannot be used on elements in arrays")
+            ctx.create_variable(target.id, value)
+            var = value
+        elif is_local and not ctx.is_var_declared(target.id):
+            var = cast_type(value, annotation)
+            var = ti.expr_init(var)
+            ctx.create_variable(target.id, var)
+        else:
+            var = target.ptr
+            if str(var.ptr.get_ret_type()) != annotation:
```

Comment:
1. Why do we use `str` here?
2. Could you also add a test for this invalid case?